src/main/java/jeresources/util/MobHelper.java

```diff
@@ -1,23 +1,59 @@
 package jeresources.util;
 
+import jeresources.api.conditionals.LightLevel;
 import jeresources.entry.MobEntry;
 import net.minecraft.entity.MobEntity;
+import net.minecraft.entity.boss.dragon.EnderDragonEntity;
+import net.minecraft.entity.monster.BlazeEntity;
+import net.minecraft.entity.monster.GuardianEntity;
+import net.minecraft.entity.monster.MagmaCubeEntity;
+import net.minecraft.entity.monster.MonsterEntity;
+import net.minecraft.entity.monster.PatrollerEntity;
+import net.minecraft.entity.monster.PhantomEntity;
+import net.minecraft.entity.monster.SlimeEntity;
+import net.minecraft.entity.monster.VexEntity;
+import net.minecraft.entity.monster.WitchEntity;
+import net.minecraft.entity.passive.AnimalEntity;
+import net.minecraft.entity.passive.BatEntity;
 import net.minecraft.entity.passive.SheepEntity;
+import net.minecraft.entity.passive.WaterMobEntity;
 
 public class MobHelper {
-    public static int getExpDrop(MobEntry entry) {
-        if (entry.getEntity() instanceof MobEntity)
-            return ((MobEntity) entry.getEntity()).experienceValue;
+    public static int getExpDrop(MobEntry entry, boolean max) {
+        if (entry.getEntity() instanceof MobEntity) {
+            int experience = ((MobEntity)entry.getEntity()).experienceValue;
+
+            if (experience != 0)
+                return experience;
+            else if (entry.getEntity() instanceof AnimalEntity || entry.getEntity() instanceof WaterMobEntity)
```

Comment:
Would put these types of if elsing in the minecraft compat code.

Suggestion: override CompatBase.registerMob in Minecraft compat, add some setters for the lightlevel and min, max exp to MobEntry and something like RenderHooks for exp and lightlevel.

Something along the lines of:
```Java
protected void registerMob(MobEntry entry) {
    ExpMapping.remap(entry);
    LightLevelMapping.remap
    super.registerMob(entry);
}
```

Remap function holds a list Functions (Java 8 ones) and just checks in a map for the class of the entity and applies the right exp and level (similar to the renderhook).

Hope this makes sense, contact me on discord if it doesn't. (way2muchnoise#9001)

Reply:
Could also be done with just a map lookup am I thinking of now. Not to over engineer with the J8 Function stuff